iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/utils/CommandDetector.java

```diff
@@ -77,6 +104,69 @@ public List<Command> search(List<ArgumentResolution> resolvedArguments) {
     return commands;
   }
 
+  /**
+   * Perform the same {@link #search(List)} but before it split the command with {@link #splitCommands(List)}.
+   */
+  public List<Command> searchWithSplit(List<ArgumentResolution> resolvedArguments) {
+    SeparatedList<List<ArgumentResolution>, String> splitCommands = splitCommands(resolvedArguments);
+    List<Command> commands = new ArrayList<>();
+
+    for (List<ArgumentResolution> resolved : splitCommands.elements()) {
+      commands.addAll(search(resolved));
+    }
+    return commands;
+  }
+
+  /**
+   * Split commands by separators: {@code &&}, {@code ||}, {@code &}, {@code |} and {@code ;}.
+   */
+  public static SeparatedList<List<ArgumentResolution>, String> splitCommands(List<ArgumentResolution> resolvedArguments) {
```

Comment:
Yes, we have e.g., https://sonarsource.atlassian.net/browse/SONARIAC-747 in mind, where we have to check only the first and the last command. Thanks to that we can first call `splitCommands()` and then `search/searchWithoutSplit` to avoid unnecessary second splitting.